gemma/gemma.cc

```diff
@@ -889,8 +889,9 @@ Gemma::Gemma(GemmaTokenizer&& tokenizer, Model model_type, Type weight_type,
       tokenizer_(std::move(tokenizer)),
       model_type_(model_type),
       weight_type_(weight_type) {
-  weights_u8_ =
-      CallForModelAndWeight<AllocateWeightsF>(model_type, weight_type, pool);
+  HWY_ASSERT(weight_type == Type::kF32);
```

Comment:
Any reason not to support other types here? We could just CallForModelAndWeight, right?

Reply:
Yeah, I understand. Just thought we could move the assert to the training code, because it seems this ctor might be useful for other weight types also.